### DISCUSSION TITLE
Updating systemd docker containers

### DISCUSSION THREAD

**Discussion by milindpatel63** [containrrr/watchtower#1100] [Q&A] [answered]
I am currently using systemd to start and stop the containers automatically if they are stoppped.
For creating, i have portainer stacks.
Is it possible to specify "sudo systemctl stop container1" command in lifecycle hooks sh script?
Are the sh scripts executed as root or user?

**Comment by milindpatel63** [comment, marked as answer]
nevermind...figured it out.
For anyone else stumbling here to find a way, here's what i did
systemd is running docker start and stop commands and is set to RestartSec=60s
that gives enough time for watchtower to update any container
i have set the following flag in watchtower WATCHTOWER_ROLLING_RESTART=true
so, watchtower pulls a new image, stops a container, and restarts it automatically.
then after 60 sec, systemd executes "ExecStart=docker start -a container" and hooks onto the running container...there's no problem if the container is running by watchtower.

**Reply by simonaronsson** [reply]
While I'm super-happy you found a working solution, I just have to ask: Is there a specific reason you opted for doing this with systemd instead of using [the built-in support for restart policies](https://docs.docker.com/config/containers/start-containers-automatically/)?

**Reply by dersimn** [reply]
There might be a couple of reasons, but mostly because your container depends on some resource outside of the Docker world, for e.g. "start only if `/myspace` is a valid mount point" (because your Docker Container mounts `/myspace`).

tbh, Watchtower would work best with SystemD if there was an option so that Watchtower only 1) Pulls the new Image 2) Stops the old container, because then any external manager would try to restart it anyways.
